### PR TITLE
VACMS-7878: Offices - Search filter input

### PR DESCRIPTION
**Pull request by allisonlu**
## Description
- Created a SearchFilter component (a text input field that acts as a filter as user is typing their keyword). 
- Created an OfficeList component so it's separate from the main App file
- Added Back to Top component

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/7878

## Testing done
Local testing done, see demo below:
https://www.loom.com/share/7aeb9313d85d41888aa65b3228478dcd

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/office-directory/components/OfficeList.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        {office.reverseFieldParentOfficeNode.entities.length > 0 ? (
+          <ChildrenOfficeList
+            offices={office.reverseFieldParentOfficeNode.entities}
+          />
+        ) : null}
+      </>
+    ));
+  }
+
+  return <h1>No results found!</h1>;
```
Comment (ryguyk) [inline src/applications/office-directory/components/OfficeList.jsx:20]: Should this be `<h1>`?

File: src/applications/office-directory/containers/App.jsx [hunk 2]
```diff
@@ -50,47 +27,22 @@ export default function App() {
...
-  const toggleFilterTopOffices = useCallback(() => {
-    setFilterTopOffices(prevValue => !prevValue);
-  }, []);
-
   return (
     <div className="vads-l-grid-container vads-u-padding-x--1p5 vads-u-padding-bottom--4">
       <h1 className="vads-u-margin-bottom--0">All VA offices</h1>
 
-      <input
+      {/* <input
```
Comment (ryguyk) [inline src/applications/office-directory/containers/App.jsx:34]: This commented code can be removed. I hope to never see this code of mine back in here!

**Review by ryguyk — APPROVED** [review]
I left just a couple comments.  I wonder if  `<h1>` is the right element for "No results", but I think we'll iron out this type of concern before we launch.

File: src/applications/office-directory/components/OfficeList.jsx (hunk 1 above)
Comment (allisonlu) [reply src/applications/office-directory/components/OfficeList.jsx:20]: Oof, thanks for catching that!

**Review by theodur — APPROVED** [review]

File: src/applications/office-directory/components/SearchFilter.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  return (
+    <div className="container">
+      <p>Search for an office, organization, department, or program.</p>
+      <label htmlFor="query" className="usa-sr-only">
+        Search:
+      </label>
+      <div className="va-flex">
+        <input
```
Comment (timwright12) [inline src/applications/office-directory/components/SearchFilter.jsx:42]: Missing ID `query` to pair with the `htmlFor` attribute in the label

**Review by timwright12 — CHANGES_REQUESTED** [review]

**Review by timwright12 — APPROVED** [review]

**Pull request merged**